### PR TITLE
Updates to ESAPI

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**

### CODE REVIEW

File: src/tss2-esys/esys_iutil.c [hunk 1]
```diff
@@ -631,50 +664,58 @@ iesys_decrypt_param(ESYS_CONTEXT * esys_context,
...
 
-            int hi = 0;
-            for (int j = 0; j < 3; j++) {
-                if (rsrc_session->authHash == rp_hash_tab[j].alg) {
-                    hi = j;
-                    break;
-                }
+        IESYS_SESSION *rsrc_session = &session->rsrc.misc.rsrc_session;
+        if (rsrc_session->type_policy_session == POLICY_PASSWORD) {
+            /* A policy password session has not auth value */
```
Comment (danintel) [inline src/tss2-esys/esys_iutil.c]: s/not auth/no auth/

File: src/tss2-esys/esys_iutil.c [hunk 2]
```diff
@@ -631,50 +664,58 @@ iesys_decrypt_param(ESYS_CONTEXT * esys_context,
...
+                                  &rp_hash_tab[hi].digest[0],
+                                  rp_hash_tab[hi].size,
+                                  &rsrc_session->nonceTPM,
+                                  &rsrc_session->nonceCaller, NULL, NULL,
+                                  rspAuths->auths[i].sessionAttributes,
+                                  &rp_hmac);
+        return_if_error(r, "HMAC error");
+
+        if (!cmp_TPM2B_AUTH(&rspAuths->auths[i].hmac, &rp_hmac)) {
+            LOG_ERROR("Invalid in TPM's response for session %i", i);
```
Comment (danintel) [inline src/tss2-esys/esys_iutil.c]: s/Invalid in TPM's/Invalid TPM/

or you can word it as:
"TPM's response is invalid for session %i"

File: test/integration/esys-create-session-auth.int.c [hunk 3]
```diff
@@ -42,6 +42,8 @@
  * key as parent key (Esys_Create).
  * The type of encryptin can be selected by the compiler variables (-D option):
  * TEST_XOR_OBFUSCATION or TEST_AES_ENCRYPTION.
+ * Secret exchange with a ECC key can be activated with the copiler variable
```
Comment (danintel) [inline test/integration/esys-create-session-auth.int.c]: s/copiler/compiler/

File: test/integration/esys-create-session-auth.int.c [hunk 4]
```diff
@@ -71,15 +73,14 @@ test_invoke_esapi(ESYS_CONTEXT * esys_context)
...
+    TPM2B_PUBLIC inPublicEcc = {
         .size = 0,
         .publicArea = {
             .type = TPM2_ALG_ECC,
             .nameAlg = TPM2_ALG_SHA256,
             .objectAttributes = (TPMA_OBJECT_USERWITHAUTH |
-                                 TPMA_OBJECT_USERWITHAUTH |
-                                 TPMA_OBJECT_RESTRICTED |
-                                 TPMA_OBJECT_SIGN_ENCRYPT |
+                                 // TPMA_OBJECT_RESTRICTED |
```
Comment (danintel) [inline test/integration/esys-create-session-auth.int.c]: Was this left here for debugging?

File: src/tss2-esys/esys_iutil.c (hunk 2 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_iutil.c]: done

File: test/integration/esys-create-session-auth.int.c (hunk 3 above)
Comment (AndreasFuchsSIT) [reply test/integration/esys-create-session-auth.int.c]: done

File: test/integration/esys-create-session-auth.int.c (hunk 4 above)
Comment (AndreasFuchsSIT) [reply test/integration/esys-create-session-auth.int.c]: done

**Comment by AndreasFuchsSIT** [comment]
all comments addresses => merging...

**Pull request merged**